raiden/network/transport/udp/udp_transport.py

```diff
@@ -543,7 +545,20 @@ def receive_delivered(self, delivered: Delivered):
         protocol, but it's required by this transport to provide the required
         properties.
         """
-        processed = ReceiveDelivered(delivered.delivered_message_identifier)
+        for queue_identifier, events in self._queueids_to_queues.items():
+            if delivered.sender != queue_identifier.recipient:
+                continue
+            if any(delivered.sender == event.recipient for event in events):
+                break
+        else:
+            log.debug(
+                'Delivered message unknown',
+                sender=pex(delivered.sender),
+                message=delivered,
+            )
+            return
+
+        processed = ReceiveDelivered(delivered.sender, delivered.delivered_message_identifier)
         self.raiden.handle_state_change(processed)
 
         message_id = delivered.delivered_message_identifier
```

Comment:
Here, if `ReceveDelivered` goes in `on_message`, this second part is called only if the return was truty